functests/validation/validation.go

```diff
@@ -0,0 +1,232 @@
+	if performanceNodeSelector == "" {
+		performanceNodeSelector = "node-role.kubernetes.io/worker-cnf="
+	}
+}
+
+var _ = Describe("validation", func() {
+	Context("General", func() {
+		It("should report all machine config pools are in ready status", func() {
+			mcp := &clientmachineconfigv1.MachineConfigPoolList{}
+			err := client.Client.List(context.TODO(), mcp, &goclient.ListOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			for _, mcItem := range mcp.Items {
+				Expect(mcItem.Status.MachineCount).To(Equal(mcItem.Status.ReadyMachineCount))
+			}
+		})
+
+		It("should have one machine config pool with the requested label", func() {
+			mcp := &clientmachineconfigv1.MachineConfigPoolList{}
+			selector, err := labels.Parse(machineConfigSelector)
+			Expect(err).ToNot(HaveOccurred())
+
+			err = client.Client.List(context.TODO(), mcp, &goclient.ListOptions{LabelSelector: selector})
+			Expect(err).ToNot(HaveOccurred())
+			Expect(len(mcp.Items)).To(Equal(1))
+		})
+
+		It("should have a openshift-ovn-kubernetes namespace", func() {
+			_, err := client.Client.Namespaces().Get("openshift-ovn-kubernetes",metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			_, err = client.Client.Namespaces().Get("openshift-sdn",metav1.GetOptions{})
+			Expect(err).To(HaveOccurred())
+			Expect(errors.IsNotFound(err)).To(BeTrue())
+		})
+	})
+
+	Context("Performance", func() {
+		getPerformanceProfiles := func() []perfv1alpha1.PerformanceProfile {
+			perfProfile := &perfv1alpha1.PerformanceProfileList{}
+			err := client.Client.List(context.TODO(), perfProfile)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(len(perfProfile.Items)).To(BeNumerically(">", 0))
+
+			return perfProfile.Items
+		}
+
+		It("should have the performance operator namespace", func() {
+			_, err := client.Client.Namespaces().Get(utils.PerformanceOperatorNamespace, metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		It("should have the performance operator deployment in running state", func() {
+			deploy, err := client.Client.Deployments(utils.PerformanceOperatorNamespace).Get(utils.PerformanceOperatorDeploymentName, metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			Expect(deploy.Status.Replicas).To(Equal(deploy.Status.ReadyReplicas))
+
+			pods, err := client.Client.Pods(utils.PerformanceOperatorNamespace).List(metav1.ListOptions{
+				LabelSelector: fmt.Sprintf("name=%s", utils.PerformanceOperatorDeploymentName)})
+			Expect(err).ToNot(HaveOccurred())
+
+			Expect(len(pods.Items)).To(Equal(1))
+			Expect(pods.Items[0].Status.Phase).To(Equal(corev1.PodRunning))
+		})
+
+		It("Should have the performance CRD available in the cluster", func() {
+			crd := &apiext.CustomResourceDefinition{}
+			err := client.Client.Get(context.TODO(), goclient.ObjectKey{Name: utils.PerformanceCRDName}, crd)
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		It("should have a performance policy applied in the cluster", func() {
```

Comment:
I would remove this. We are going to validate the cluster before running tests. The performance profile is going to be applied later, as much as sriov networks / policies.